crates/control/src/motion/step_planner.rs

```diff
@@ -36,15 +37,234 @@ pub struct CycleContext {
+        let support_side = match context.support_foot.support_side {
+            Some(side) => side,
+            None => {
+                return Ok(MainOutputs {
+                    step_plan: Step {
+                        forward: 0.0,
+                        left: 0.0,
+                        turn: 0.0,
+                    }
+                    .into(),
+                })
+            }
+        };
+
+        let (path, orientation_mode) = match context.motion_command {
+            MotionCommand::Walk {
+                path,
+                orientation_mode,
+                ..
+            } => (path, orientation_mode),
+            _ => {
+                return Ok(MainOutputs {
+                    step_plan: Step {
+                        forward: 0.0,
+                        left: 0.0,
+                        turn: 0.0,
+                    }
+                    .into(),
+                })
+            }
+        };
+
+        let segment = path
+            .iter()
+            .scan(0.0f32, |distance, segment| {
+                let result = if *distance < context.max_step_size.forward {
+                    Some(segment)
+                } else {
+                    None
+                };
+                *distance += segment.length();
+                result
+            })
+            .last()
+            .ok_or_else(|| eyre!("empty path provided"))?;
+
+        let target_pose = match segment {
+            PathSegment::LineSegment(line_segment) => {
+                let direction = line_segment.1;
+                let rotation = if direction.coords.norm_squared() < f32::EPSILON {
+                    UnitComplex::identity()
+                } else {
+                    UnitComplex::from_cos_sin_unchecked(direction.x, direction.y)
+                };
+                Isometry2::from_parts(line_segment.1.into(), rotation)
+            }
+            PathSegment::Arc(arc, orientation) => {
+                let direction = orientation
+                    .rotate_vector_90_degrees(arc.start - arc.circle.center)
+                    .normalize();
+                Isometry2::from_parts(
+                    (arc.start + direction * context.max_step_size.forward).into(),
+                    UnitComplex::from_cos_sin_unchecked(direction.x, direction.y),
+                )
+            }
+        };
+
+        let mut step = Step {
+            forward: target_pose.translation.x,
+            left: target_pose.translation.y,
+            turn: match orientation_mode {
+                OrientationMode::AlignWithPath => target_pose.rotation,
+                OrientationMode::Override(orientation) => *orientation,
+            }
+            .angle(),
+        };
+
+        if let Some(injected_step) = context.injected_step {
+            step = *injected_step;
+        }
+
+        let step = compensate_with_return_offset(step, *context.walk_return_offset);
+
+        // clamp_step_to_walk_volume
```

Comment:
These comments are redundant to the called function. Remove? Same below.

Reply:
yep, done